cmd/server/cadence/fx.go

```diff
@@ -32,23 +32,46 @@ import (
 	"github.com/uber/cadence/common/config"
 	"github.com/uber/cadence/common/dynamicconfig"
 	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/membership/membershipfx"
+	"github.com/uber/cadence/common/peerprovider/ringpopprovider/ringpopfx"
 	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin/cassandra/gocql"
+	"github.com/uber/cadence/common/rpc/rpcfx"
+	"github.com/uber/cadence/common/service"
+	"github.com/uber/cadence/service/sharddistributor/sharddistributorfx"
 	"github.com/uber/cadence/tools/cassandra"
 	"github.com/uber/cadence/tools/sql"
 )
 
 // Module provides a cadence server initialization with root components.
 // AppParams allows to provide optional/overrides for implementation specific dependencies.
-var Module = fx.Options(
-	fx.Provide(NewApp),
-	// empty invoke so fx won't drop the application from the dependencies.
-	fx.Invoke(func(a *App) {}),
-)
+func Module(serviceName string) fx.Option {
+	if serviceName == service.ShortName(service.ShardDistributor) {
+		return fx.Options(
+			fx.Supply(serviceContext{
+				Name:     serviceName,
+				FullName: service.FullName(serviceName),
+			}),
+			rpcfx.Module,
+			// PeerProvider could be overriden. Inside Uber we use DNS based internal solution.
```

Comment:
I don't think we should mention Ubers internal needs directly here, maybe
```suggestion
			// PeerProvider could be overriden e.g. with a DNS based solution.
```